src/olympia/yara/tasks.py

```diff
@@ -5,17 +5,18 @@
 
 import olympia.core.logger
 
-from olympia.amo.celery import task
+from olympia.devhub.tasks import validation_task
 from olympia.files.models import FileUpload
 from olympia.files.utils import SafeZip
 
+
 from .models import YaraResult
 
 log = olympia.core.logger.getLogger('z.yara.task')
 
 
-@task
-def run_yara(upload_pk):
+@validation_task
+def run_yara(results, upload_pk):
```

Comment:
Since the docstring mentions storing the `results` and it now, as a `validation_task` receives a `results` argument (and returns it) it may start to be a little confusing... I don't have good suggestions though, just something to keep in mind maybe.

Reply:
Ah, good catch, thanks. I'll revisit the docstring.